logstash-core/lib/logstash/java_pipeline.rb

```diff
@@ -548,6 +548,10 @@ def inspect
     }
   end
 
+  def shutdown_requested
```

Comment:
```suggestion
  def shutdown_requested?
```

Reply:
Also, while this is the _only_ effectively-concrete implementation of `JavaBasePipelineExt`, we use `JavaBasePipelineExt` in several sections of the java code for typesafety, so we should at least add the _interface_ for this to the superclass (even if the superclass is not capable of implementing it). We should be able to add something like the following to `JavaBasePipelineExt`'s definition:

~~~ java
    @JRubyMethod(name = "shutdown_requested?")
    public IRubyObject isShutdownRequested(final ThreadContext context) {
        throw new IllegalStateException("Pipeline implementation does not provide `shutdown_requested?`, which is a Logstash internal error.");
    }
~~~

(the magic renaming [Beans Conventions](https://github.com/jruby/jruby/wiki/CallingJavaFromJRuby#alternative-names-and-beans-convention) makes the subclass-in-ruby's `shutdown_requested?` work with the superclass-in-java's `isShutdownRequested`)

Reply:
The only subclass of `JavaBasePipelineExt` is  `JavaPipeline`. Can you tell me the benefit of adding this interface? What makes `shutdown_requested?` special to be in superclass? I am lost here

Reply:
It will work as-is, but it is fragile to future changes. 

The `ExecutionContext#pipeline` effectively is type-safety bound to be a `JavaBasePipelineExt` because of how it is instantiated/populated in java code, so while `JavaPipeline` is effectively (currently) the only instantiable implementation, the compiler doesn't know that.

Adding it to the superclass ensures that if and when we implement a new subclass, the requirements that the plugins have on it are a part of the _contract_ instead of a surprise `NoMethodError` at execution time.